File: python/数组/1228. 等差数列中缺失的数字.py
from typing import List
class Solution:
    def missingNumber(self, arr: List[int]) -> int:
        d = (arr[-1]-arr[0])//len(arr)
        i = 0
        while i < len(arr)-1:
            if arr[i] + d != arr[i+1]:
                return arr[i]+d
            i+=1

File: go/math/1228. 等差数列中缺失的数字.go
package math

func missingNumber(arr []int) int {
	d := (arr[len(arr)-1] - arr[0]) / len(arr)
	if d == 0 {
		return arr[0]
	}
	for i := 0; i < len(arr)-1; i++ {
		if arr[i]+d != arr[i+1] {
			return arr[i] + d
		}
	}
	return d
}
